test/libsolidity/semanticTests/functionTypes/inline_array_payable_functions.sol

```diff
@@ -0,0 +1,18 @@
+contract C {
+    function f() external payable returns (uint) { assert(msg.value > 0); return 1; }
+    function g() external payable returns (uint) { assert(msg.value > 0); return 2; }
+
+    function h() public returns (uint) {
+        assert((true ? this.f{value: 1} : this.g{value: 1})() == 1);
+        assert((false ? this.f : this.g){value: 1}() == 2);
+
+        assert([this.f, this.g][0]{value: 1}() == 1);
```

Comment:
All of these fail as there are some stack issues.
Which one of these we actually want to work properly? @chriseth 

First one should probably fail compilation same like for inline array.

Reply:
I think it's not worth trying to support them - we also do not support such types in memory.

The fix could be to return `nullptr` in `mobileType()` if value or gas is set. Should probably do the same for bound and saltSet. If possible, we could also add an assertion about `m_kind` being internal, external or declaration.

Reply:
I agree. It would take too much effort to support these.

One part of solution is to return `nullptr` as you said. That solves getting mobile function types for functions with value set.
Second part is when value is associated with call itself, like this:
`(false ? this.f : this.g){value: 1}();`

Reply:
The "second part" should work - it should be the same as external function pointers.